newrelic/resource_newrelic_synthetics_monitor.go

```diff
@@ -95,158 +94,411 @@ func resourceNewRelicSyntheticsMonitor() *schema.Resource {
+	return nil
 }
 
 func resourceNewRelicSyntheticsMonitorRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*ProviderConfig).NewClient
 
 	log.Printf("[INFO] Reading New Relic Synthetics monitor %s", d.Id())
 
-	monitor, err := client.Synthetics.GetMonitorWithContext(ctx, d.Id())
+	resp, err := client.Entities.GetEntityWithContext(ctx, common.EntityGUID(d.Id()))
+
 	if err != nil {
 		if _, ok := err.(*errors.NotFound); ok {
 			d.SetId("")
 			return nil
 		}
-
 		return diag.FromErr(err)
 	}
 
-	readSyntheticsMonitorStruct(monitor, d)
+	setCommonSyntheticsMonitorAttributes(resp, d)
 
 	return nil
 }
 
+//func to set output values in the read func.
+func setCommonSyntheticsMonitorAttributes(v *entities.EntityInterface, d *schema.ResourceData) {
+
+	switch e := (*v).(type) {
+
+	case *entities.SyntheticMonitorEntity:
+		_ = d.Set("name", e.Name)
+		_ = d.Set("type", e.MonitorType)
+		_ = d.Set("uri", e.MonitoredURL)
+
+	}
+
+}
+
+//func to build input to update simple browser monitor
+func buildSyntheticsSimpleBrowserMonitorUpdateStruct(d *schema.ResourceData) synthetics.SyntheticsUpdateSimpleBrowserMonitorInput {
+
+	simpleBrowserMonitorUpdateInput := synthetics.SyntheticsUpdateSimpleBrowserMonitorInput{}
+
+	inputBase := expandSyntheticsMonitorBase(d)
+
+	simpleBrowserMonitorUpdateInput.AdvancedOptions.CustomHeaders = inputBase.CustomHeaders
+	simpleBrowserMonitorUpdateInput.Name = inputBase.Name
+	simpleBrowserMonitorUpdateInput.Period = inputBase.Period
+	simpleBrowserMonitorUpdateInput.Status = inputBase.Status
+	simpleBrowserMonitorUpdateInput.Tags = inputBase.Tags
+	simpleBrowserMonitorUpdateInput.Uri = inputBase.URI
+
+	if v, ok := d.GetOk("locations_public"); ok {
+		simpleBrowserMonitorUpdateInput.Locations.Public = expandSyntheticsSimplePublicLocations(v.(*schema.Set).List())
+	}
+
+	if v, ok := d.GetOk("locations_private"); ok {
+		simpleBrowserMonitorUpdateInput.Locations.Private = expandSyntheticsSimplePrivateLocations(v.(*schema.Set).List())
+	}
+
+	if v, ok := d.GetOk("enable_screenshot_on_failure_and_script"); ok {
+		simpleBrowserMonitorUpdateInput.AdvancedOptions.EnableScreenshotOnFailureAndScript = v.(bool)
+	}
+
+	if v, ok := d.GetOk("validation_string"); ok {
+		simpleBrowserMonitorUpdateInput.AdvancedOptions.ResponseValidationText = v.(string)
+	}
+
+	if v, ok := d.GetOk("verify_ssl"); ok {
+		simpleBrowserMonitorUpdateInput.AdvancedOptions.UseTlsValidation = v.(bool)
+	}
+
+	if v, ok := d.GetOk("script_language"); ok {
+		simpleBrowserMonitorUpdateInput.Runtime.ScriptLanguage = v.(string)
+	}
+
+	if v, ok := d.GetOk("runtime_type"); ok {
+		simpleBrowserMonitorUpdateInput.Runtime.RuntimeType = v.(string)
+	}
+
+	if v, ok := d.GetOk("runtime_type_version"); ok {
+		simpleBrowserMonitorUpdateInput.Runtime.RuntimeTypeVersion = synthetics.SemVer(v.(string))
+	}
+
+	return simpleBrowserMonitorUpdateInput
+}
+
+//func to build input to update simple monitor
```

Comment:
Take a look at comments in `newrelic/resource_newrelic_synthetics_script_monitor.go`. We are using different comment types throughout this PR. We should be constant.